examples/rust/sysvar/src/processor.rs

```diff
@@ -43,3 +43,98 @@ pub fn process_instruction(
 
     Ok(())
 }
+
+
+//Code for deserializing the SlotHash sysvar while remaining within the BPF compute limit 
+
+//entrypoint!(process_instruction2);
+fn process_instruction2(
+    _program_id: &Pubkey,
+    accounts: &[AccountInfo],
+    _instruction_data: &[u8],
+) -> ProgramResult {
+    let accounts_iter = &mut accounts.iter();
+    let sysvar_slot_history = next_account_info(accounts_iter)?;
+
+    /*
+        Decoding the SlotHashes sysvar using `from_account_info` is too expensive.
+        For example this statement will exceed the current BPF compute unit budget:
+
+            let slot_hashes = SlotHashes::from_account_info(&sysvar_slot_history).unwrap();
+
+        Instead manually decode the sysvar.
+    */
+
+    if *sysvar_slot_history.key != sysvar::slot_hashes::id() {
+        msg!("Invalid SlotHashes sysvar");
+        return Err(ProgramError::InvalidArgument);
+    }
+
+    let data = sysvar_slot_history.try_borrow_data()?;
+
+    let num_slot_hashes = u64::from_le_bytes(data[0..8].try_into().unwrap());
+    let mut pos = 8;
+
+    for _i in 0..num_slot_hashes {
+        let slot = u64::from_le_bytes(data[pos..pos + 8].try_into().unwrap());
+        pos += 8;
+        let hash = &data[pos..pos + 32];
+        pos += 32;
+
+        if slot == 54943128 {
+            msg!("Found slot {}, hash {}", slot, Hash::new(hash));
+        }
+    }
+
+    Ok(())
+}
+
+#[cfg(test)]
+mod test {
+    use {
+        super::*,
+        assert_matches::*,
+        solana_program::{
+            instruction::{AccountMeta, Instruction},
+            native_token::sol_to_lamports,
+            sysvar,
+        },
+        solana_program_test::*,
+        solana_sdk::{signature::Signer, transaction::Transaction},
+    };
+
+    #[tokio::test]
+    async fn test_transaction() {
```

Comment:
The idea is to merge this test with the one in  `examples/rust/sysvar/tests/functional.rs` since they exercise the same code underneath.  You can change the transaction to also include the slot hashes sysvar and rest should work very nicely